cluster-install/install.sh

```diff
@@ -2131,8 +2172,13 @@ step_apply_resources() {
 step_apply_trident_operator_patches() {
     logheader "$__DEBUG" "$(prefix_dryrun "Applying Trident Operator patches...")"
     local -ra patches=("${_PATCHES_TRIDENT_OPERATOR[@]}")
+    local -r patches_len="${#patches[@]}"
 
-    if [ "$LOG_LEVEL" == "$__DEBUG" ] && [ "${#patches[@]}" -gt 0 ]; then
+    if ! trident_will_be_installed_or_modified && [ "$patches_len" -gt 0 ]; then
+        fatal "found $patches_len operator patches (expected 0) despite trident not being installed or modified"
```

Comment:
Another guardrail to make sure we don't end up generating patches for components that aren't being acted upon.